arduino-ide-extension/src/browser/widgets/sketchbook/sketchbook-tree-widget.tsx

```diff
@@ -79,29 +110,33 @@ export class SketchbookTreeWidget extends FileTreeWidget {
         };
     }
 
-    protected renderInlineCommands(node: TreeNode, props: NodeProps): React.ReactNode {
-        if (SketchbookTree.SketchDirNode.is(node) && node.commands && node.id === this.hoveredNodeId) {
+    // TODO: fix node "any" type
+    protected renderInlineCommands(node: any, props: NodeProps): React.ReactNode {
+
+        if (SketchbookTree.SketchDirNode.is(node) && (node.commands && node.id === this.hoveredNodeId || this.currentSketchUri === node?.uri.toString())) {
             return Array.from(new Set(node.commands)).map(command => this.renderInlineCommand(command.id, node));
         }
         return undefined;
     }
 
     protected renderInlineCommand(commandId: string, node: SketchbookTree.SketchDirNode): React.ReactNode {
         const command = this.commandRegistry.getCommand(commandId);
-        const icon = command?.iconClass;
         const args = { model: this.model, node: node };
-        if (command && icon && this.commandRegistry.isEnabled(commandId, args) && this.commandRegistry.isVisible(commandId, args)) {
-            const className = [TREE_NODE_SEGMENT_CLASS, TREE_NODE_TAIL_CLASS, icon, 'theia-tree-view-inline-action'].join(' ');
-            return <div
+        if (command && this.commandRegistry.isEnabled(commandId, args) && this.commandRegistry.isVisible(commandId, args)) {
+            const className = [TREE_NODE_SEGMENT_CLASS, TREE_NODE_TAIL_CLASS, 'theia-tree-view-inline-action'].join(' ');
+            return (<div
                 key={`${commandId}--${node.id}`}
                 className={className}
                 title={command?.label || command.id}
-                onClick={event => {
+                onClick={(event) => {
                     event.preventDefault();
                     event.stopPropagation();
-                    this.commandRegistry.executeCommand(commandId, Object.assign(args, { event: event.nativeEvent }))
-                }}
-            />;
+                    this.commandRegistry.executeCommand(commandId, Object.assign(args, { event: event.nativeEvent }));
+                }
+                }
+            >
+                <div className='sketchbook-tree__opts' />
```

Comment:
How can I add a custom command? If I add a random command with an icon to the tree node I see the ellipses twice but not my command.

![Screen Shot 2021-04-28 at 10 33 13](https://user-images.githubusercontent.com/1405703/116372903-255fab00-a80d-11eb-80f7-f427bad882ac.jpg)

Reply:
your command should still work, the issue is that I force the ellipsis, which is deeply wrong. I should have the icon associated to the command itself, and then use that.
I'm not sure I can do that with non-default icons... let me dig into that.

thanks!

Reply:
should be fixed :)